torch/distributed/device_mesh.py

```diff
@@ -434,6 +440,23 @@ def get_group(
                     )
                 return dim_groups
 
+        @staticmethod
+        def from_group(group: ProcessGroup, device_type: str) -> "DeviceMesh":
```

Comment:
can we add some documentation?

Also I am thinking how we should support multi-dim from group, i.e. HSDP shard, replicate group to pass in to construct a DeviceMesh, I guess the `group` argument would likely need to be `Union[ProcessGroup, List[ProcessGroup]]`

Reply:
> Also I am thinking how we should support multi-dim from group, i.e. HSDP shard, replicate group to pass in to construct a DeviceMesh, I guess the group argument would likely need to be Union[ProcessGroup, List[ProcessGroup]]

This makes sense to me. Shall we expand the API later when we need it in a separate PR?

Reply:
yeah we don't need to do it in this PR as it's BC compatible if later we expand the API